packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -86,94 +308,44 @@ def create_client(
-
-    if all:
-        measurement_service_class = get_all_registered_measurement_service_classes(discovery_client)
-        if len(measurement_service_class) == 0:
-            raise click.ClickException("No registered measurements.")
-    else:
-        if not measurement_service_class:
-            raise click.ClickException(
-                "The measurement service class cannot be empty. Either provide a measurement service class or use the 'all' flag to generate clients for all registered measurements."
-            )
 
-    if directory_out is None:
-        directory_out_path = pathlib.Path.cwd()
+    has_batch_parameters = (
+        measurement_service_class or all or module_name or class_name or directory_out
+    )
+    if has_batch_parameters and not interactive:
+        _create_client(
+            channel_pool=channel_pool,
+            discovery_client=discovery_client,
+            measurement_service_class=measurement_service_class,
+            all=all,
+            module_name=module_name,
+            class_name=class_name,
+            directory_out=directory_out,
+        )
     else:
-        directory_out_path = pathlib.Path(directory_out)
-
-    if not directory_out_path.exists():
-        raise click.ClickException(f"The specified directory '{directory_out}' was not found.")
-
-    is_multiple_client_generation = len(measurement_service_class) > 1
-    for service_class in measurement_service_class:
-        if is_multiple_client_generation or module_name is None or class_name is None:
-            base_service_class = service_class.split(".")[-1]
-            base_service_class = remove_suffix(base_service_class)
-
-            if not base_service_class.isidentifier():
-                raise click.ClickException(
-                    "Client creation failed.\nEither provide a module name or update the measurement with a valid service class."
-                )
-            if not any(ch.isupper() for ch in base_service_class):
-                print(
-                    f"Warning: The service class '{service_class}' does not adhere to the recommended format."
-                )
-
-            if is_multiple_client_generation:
-                module_name = camel_to_snake_case(base_service_class) + "_client"
-                class_name = base_service_class.replace("_", "") + "Client"
-            else:
-                if module_name is None:
-                    module_name = camel_to_snake_case(base_service_class) + "_client"
-                if class_name is None:
-                    class_name = base_service_class.replace("_", "") + "Client"
-
-        if not is_python_identifier(module_name):
+        if has_batch_parameters and interactive:
             raise click.ClickException(
-                f"The module name '{module_name}' is not a valid Python identifier."
+                "Interactive mode does not support additional parameters. Please remove any extra parameters and try again."
             )
```

Comment:
Should we consider following an early exit pattern and check for this exception before executing anything else?

Reply:
With the usage of click-option-group, interactive mode is mutually exclusive of batch mode. So, I have removed this code.